### PR TITLE
feat: add inclusive for iter until

### PR DESCRIPTION
**Pull request by qazxcdswe123**
Added `inclusive` function for later `..=` and `..<`

### CODE REVIEW

File: builtin/iter_test.mbt [hunk 1]
```diff
@@ -259,6 +259,38 @@ test "range" {
...
+  )
+  inspect!(
+    (@int64.max_value() - 2).until(@int64.max_value() - 1L, inclusive=true),
+    content="[9223372036854775805, 9223372036854775806]",
+  )
+  inspect!(
+    (@int64.max_value() - 1).until(@int64.max_value(), inclusive=true),
+    content="[9223372036854775806, 9223372036854775807]",
+  )
+  inspect!(
```
Comment (qazxcdswe123) [inline builtin/iter_test.mbt:278]: Since the accuracy is not enough, the inclusive param doesn't affect the result here.

File: builtin/iter.mbt [hunk 2]
```diff
@@ -226,6 +226,40 @@ pub fn until(self : Int, end : Int, ~step : Int = 1) -> Iter[Int] {
...
+///
+/// # Returns
+///
+/// Returns an iterator that iterates over the range of Int from `start` to `end`.
+pub fn until_inclusive(self : Int, end : Int, ~step : Int = 1) -> Iter[Int] {
+  if step == 0 {
+    return Iter::empty()
+  }
+  fn(yield) {
+    let mut i = self
```
Comment (bobzhang) [inline builtin/iter.mbt:246]:
```
for i in a ..< b {
}
==>
    for i a.upto(b)
for i in a ..= b {
} 
==>
    for i in a.upto(b, inclusive=true)
```

File: builtin/builtin.mbti [hunk 3]
```diff
@@ -360,7 +360,8 @@ impl Int {
   to_json(Int) -> Json
   to_string(Int) -> String
   to_uint(Int) -> UInt
-  until(Int, Int, ~step : Int = ..) -> Iter[Int]
+  until(Int, Int, ~step : Int = .., ~inclusive : Bool = ..) -> Iter[Int]
```
Comment (qazxcdswe123) [inline builtin/builtin.mbti:363]: This is too repetitive though, they serve the same purpose (default `step` for `until` is 1, behave the same as `upto`). I think we can keep only `until` and desugar to it.

Comment (qazxcdswe123) [reply builtin/builtin.mbti:363]: The compiler should work just fine BTW

**Pull request merged**